### PR TITLE
pip-audit: 2.4.12 -> 2.5.4

### PR DESCRIPTION
**Pull request by ocfox**
###### Description of changes
https://github.com/pypa/pip-audit/compare/v2.4.12...v2.5.4

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by fabaff — COMMENTED** [review]
```bash
error: builder for '/nix/store/xnnvv6i42kx2fwhw1rak0p3pv7sfq716-pip-audit-2.5.4.drv' failed with exit code 1;
       last 10 log lines:
       > E               AttributeError: 'Component' object has no attribute 'add_vulnerability'
       >
       > pip_audit/_format/cyclonedx.py:37: AttributeError
       > =========================== short test summary info ============================
       > FAILED test/format/test_cyclonedx.py::test_cyclonedx_inner_json - AttributeError: 'Component' object has no attribute 'add_vulnerability'
       > FAILED test/format/test_cyclonedx.py::test_cyclonedx_inner_xml - AttributeError: 'Component' object has no attribute 'add_vulnerability'
       > FAILED test/format/test_cyclonedx.py::test_cyclonedx_skipped_dep - AttributeError: 'Component' object has no attribute 'add_vulnerability'
       > FAILED test/format/test_cyclonedx.py::test_cyclonedx_fix - AttributeError: 'Component' object has no attribute 'add_vulnerability'
       > ================= 4 failed, 91 passed, 13 deselected in 0.64s ==================
```

**Comment by ocfox** [comment]
Result of `nixpkgs-review pr 224590` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>pip-audit</li>
    <li>pip-audit.dist</li>
  </ul>
</details>

**Comment by SFrijters** [comment]
Result of `nixpkgs-review pr 224590` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>pip-audit</li>
    <li>pip-audit.dist</li>
  </ul>
</details>

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 224590` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>pip-audit</li>
    <li>pip-audit.dist</li>
  </ul>
</details>

**Comment by SuperSandro2000** [comment]
@ofborg build pip-audit

File: pkgs/development/tools/pip-audit/default.nix [hunk 1]
```diff
@@ -43,13 +40,11 @@ buildPythonApplication rec {
     cachecontrol
     cyclonedx-python-lib
     html5lib
-    lockfile
```
Comment (SuperSandro2000) [inline pkgs/development/tools/pip-audit/default.nix:46]:
```
ERROR: Could not find a version that satisfies the requirement lockfile>=0.9; extra == "filecache" (from cachecontrol[filecache]) (from versions: none)
ERROR: No matching distribution found for lockfile>=0.9; extra == "filecache"
```

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 224590` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>pip-audit</li>
    <li>pip-audit.dist</li>
  </ul>
</details>

File: pkgs/development/tools/pip-audit/default.nix (hunk 1 above)
Comment (fabaff) [reply pkgs/development/tools/pip-audit/default.nix:46]: Fixed, I committed the the wrong change.

**Pull request merged**